ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -406,205 +439,427 @@ checkKnownIntersectionInvariants cfg KnownIntersectionState
   where
     SecurityParam k = protocolSecurityParam cfg
 
-assertKnownIntersectionInvariants
-  :: ( HasHeader blk
-     , HasHeader (Header blk)
-     , HasAnnTip blk
-     , ConsensusProtocol (BlockProtocol blk)
-     , HasCallStack
-     )
-  => ConsensusConfig (BlockProtocol blk)
-  -> KnownIntersectionState blk
-  -> KnownIntersectionState blk
+    KnownIntersectionState {
+        mostRecentIntersection
+      , ourFrag
+      , theirFrag
+      , theirHeaderStateHistory
+      } = kis
+
+assertKnownIntersectionInvariants ::
+    ( HasHeader blk
+    , HasHeader (Header blk)
+    , HasAnnTip blk
+    , ConsensusProtocol (BlockProtocol blk)
+    , HasCallStack
+    )
+ => ConsensusConfig (BlockProtocol blk)
+ -> KnownIntersectionState blk
+ -> KnownIntersectionState blk
 assertKnownIntersectionInvariants cfg kis =
     assertWithMsg (checkKnownIntersectionInvariants cfg kis) kis
 
+{-------------------------------------------------------------------------------
+  The ChainSync client definition
+-------------------------------------------------------------------------------}
+
+-- | Arguments determined by configuration
+--
+-- These are available before the diffusion layer is online.
+data ConfigEnv m blk = ConfigEnv {
+    mkPipelineDecision0     :: MkPipelineDecision
+    -- ^ The pipelining decider to use after 'MsgFoundIntersect' arrives
+  , tracer                  :: Tracer m (TraceChainSyncClientEvent blk)
+  , cfg                     :: TopLevelConfig blk
+  , someHeaderInFutureCheck :: InFutureCheck.SomeHeaderInFutureCheck m blk
+  , chainDbView             :: ChainDbView m blk
```

Comment:
Despite appreciating the attempt to reduce renaming, once we are in the journey of making a breaking change I would rather prefix the fields with `ce` at least.

Reply:
In the call, we agreed to keep unprefixed names. But we're going to add a hlint rule that required qualified import and limits the names to `CSC` or `CSClient` or `CsClient`.

Reply:
I finally found the HLint manual: https://github.com/ndmitchell/hlint/blob/master/README.md#customizing-the-hints

FYI: we want `explicitOrQualified` instead of just `qualified`.

Reply:
However, it seems to not work :/ I'll open an Issue to investigate.

Reply:
Doh! Our `.gitignore` explicitly forbids `.hlint.yaml`. I don't want to shave this yak in this PR.

Reply:
I opened https://github.com/input-output-hk/ouroboros-consensus/issues/833